### ISSUE TITLE
Sequential packets and break block events

### ISSUE THREAD

**Issue by zly2006** [FabricMC/fabric#3836]
# The bug

1. cancel all AttackBlockCallback.EVENT
2. place a sign with some texts
3. break the sign
4. the sign still exists, but the texts disappeared
5. if you rejoin the game, the texts appear again

This is a bug in [vanilla minecraft](https://bugs.mojang.com/browse/MC-272862), but will fabric api fix it? All you need to do is to send a PlayerActionResponseS2CPacket before BlockUpdateS2CPacket, so the client-side block entity will not be in a removal state.

**Comment by PSJahn** [comment]
Since this is a bug in vanilla, I do not think the Fabric API should tinker with it as it shouldn't modify the behavior on its own.

**Comment by zly2006** [comment]
> Since this is a bug in vanilla, I do not think the Fabric API should tinker with it as it shouldn't modify the behavior on its own.

Nah, this is definitely a bug and it breaks many mods that try to protect some blocks from being harvested. I think it is OK for FAPI to do this caz nobody wants the sign content to disappear.

**Comment by Linguardium** [comment]
its also directly caused by cancelling block breaking via a fapi event

**Comment by zly2006** [comment]
> its also directly caused by cancelling block breaking via a fapi event

Yeah thats why I open this issue here. Once your mod uses FAPI to "protect" some blocks you will then be involved with this bug.
